--- a/trace/trace.c
+++ b/trace/trace.c
@@ -228,7 +228,7 @@ static void tracer_cb(struct uloop_proce
 			if (!tracee->in_syscall) {
 #ifdef __aarch64__
 				int syscall = -1;
-				struct ptrace_syscall_info ptsi = {.op=PTRACE_SYSCALL_INFO_ENTRY};
+				struct __ptrace_syscall_info ptsi = {.op=PTRACE_SYSCALL_INFO_ENTRY};
 				if (ptrace(PTRACE_GET_SYSCALL_INFO, c->pid, sizeof(ptsi), &ptsi) != -1)
 					syscall = ptsi.entry.nr;
 #else
@@ -262,7 +262,7 @@ static void tracer_cb(struct uloop_proce
 		} else if ((ret >> 8) == (SIGTRAP | (PTRACE_EVENT_SECCOMP << 8))) {
 #ifdef __aarch64__
 			int syscall = -1;
-			struct ptrace_syscall_info ptsi = {.op=PTRACE_SYSCALL_INFO_SECCOMP};
+			struct __ptrace_syscall_info ptsi = {.op=PTRACE_SYSCALL_INFO_SECCOMP};
 			if (ptrace(PTRACE_GET_SYSCALL_INFO, c->pid, sizeof(ptsi), &ptsi) != -1)
 				syscall = ptsi.entry.nr;
 #else
